### ISSUE TITLE
Install err with pip 7 on python 2.7 does not perform Python 2 translation

### ISSUE THREAD

**Issue by eykd** [gbin/err#469]
When installing `err` using the most recent pip (7.1.2) on Python 2.7.9, the `err` files were not converted from Py3k to Py2, resulting in import errors. Downgrading to pip 6.1.1 results in a working errbot.

I seem to get the same problem with the latest err==3.0.3, though it gives a more graceful error message:

```
You are trying to run err under python2 without converting the source code to py2 first.
Either use python3 or install err using ./setup.py develop.
```

I realize this may be a bug in pip, but I'm not sure how the translation routines work or interface w/ setuptools/pip. Hopefully it can be fixed in setup.py?

**Comment by gbin** [comment]
Some analysis...
errbot is a source package (sdist) for the obvious reason that we need to be able to convert it at the last minute.

It looks like pip now runs : setup.py bdist_wheel before installing the package

basically building a **binary** package before installing it.

**Comment by gbin** [comment]
So I have added a pip upgrade to our CI and used pip to install err instead of directly setup.py.

It looks like it works under py2.7 https://travis-ci.org/gbin/err/jobs/80021113

Can you try 3.0.4 (just released it) or head to see if you can reproduce this ?

**Comment by mrshu** [comment]
@gbin I seem to be no longer reproduce this with `3.0.4`.

**Comment by eykd** [comment]
I'll check on Monday. Thanks for the quick response!

**Comment by eykd** [comment]
In a fresh virtualenv, w/ pip 7.1.2, with a fresh installation of 3.0.4, I get this:

```
$ err.py --text
You are trying to run err under python2 without converting the source code to py2 first.
Either use python3 or install err using ./setup.py develop.
```

**Comment by mrshu** [comment]
I can reproduce this problem with the versions mentioned above. Sorry about my previous comment -- it turns out I had an outdated version of pip installed.

**Comment by gbin** [comment]
I don't get it. Are you sure you are not starting it from a git clone of err or something maybe ?

I just did this:

```
➜  ~  mkvirtualenv -p `which python2` py2  
Running virtualenv with interpreter /usr/bin/python2
New python executable in py2/bin/python2
Also creating executable in py2/bin/python
Installing setuptools, pip...done.
(py2)➜  ~  pip install --upgrade pip
Downloading/unpacking pip from https://pypi.python.org/packages/py2.py3/p/pip/pip-7.1.2-py2.py3-none-any.whl#md5=5ff9fec0be479e4e36df467556deed4d
  Downloading pip-7.1.2-py2.py3-none-any.whl (1.1MB): 1.1MB downloaded
Installing collected packages: pip
  Found existing installation: pip 1.5.4
    Uninstalling pip:
      Successfully uninstalled pip
Successfully installed pip
Cleaning up...
(py2)➜  ~  pip install err
[...]
Running setup.py install for err
Successfully installed MarkupSafe-0.23 Pygments-2.0.2 WebOb-1.4.1 ansi-0.1.3 backports.functools-lru-cache-1.0.3 beautifulsoup4-4.4.0 bottle-0.12.8 cffi-1.2.1 colorlog-2.6.0 config-0.3.9 cryptography-1.0.1 daemonize-2.3.1 dnspython-1.12.0 enum34-1.0.4 err-3.0.4 idna-2.0 ipaddress-1.0.14 jinja2-2.8 markdown-2.6.2 pyOpenSSL-0.15.1 pyasn1-0.1.8 pycparser-2.14 pygments-markdown-lexer-0.1.0.dev39 requests-2.7.0 six-1.9.0 waitress-0.8.10 webtest-2.0.18 yapsy-1.11.223

(py2)➜  ~  cp ~/.virtualenvs/py2/lib/python2.7/site-packages/errbot/config-template.py conf-py2.py
(py2)➜  ~  vi conf-py2.py   # changed the 2 paths to /tmp/ and /tmp/err.log
(py2)➜  ~  err.py -c conf-py2.py -T
11:19:15 INFO     __main__                  Config check passed...
11:19:15 INFO     __main__                  Selected backend 'Text'.
11:19:15 INFO     __main__                  Checking for '/tmp'...
11:19:15 INFO     errbot.backend_manager    Backends search paths [u'/usr/local/google/home/gbin/.virtualenvs/py2/local/lib/python2.7/site-packages/errbot/backends']
```

**Comment by mrshu** [comment]
@gbin this is what I just did:

``` 
$ virtualenv-2.7 --version
13.1.2
$ virtualenv-2.7 /tmp/py2
New python executable in /tmp/py2/bin/python2
Also creating executable in /tmp/py2/bin/python
Installing setuptools, pip, wheel...done.
$ cd /tmp/py2                                                       
$ ls
bin  include  lib
$ source bin/activate.
activate.csh   activate.fish  
$ source bin/activate
(py2)$ pip install --upgrade pip
Requirement already up-to-date: pip in ./lib/python2.7/site-packages
(py2)$ pip install err
Collecting err
/tmp/py2/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:90: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cau
se certain SSL connections to fail. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
Collecting daemonize (from err)
Requirement already satisfied (use --upgrade to upgrade): setuptools in ./lib/python2.7/site-packages (from err)
Collecting pygments-markdown-lexer>=0.1.0.dev39 (from err)
Collecting pyOpenSSL (from err)
  Using cached pyOpenSSL-0.15.1-py2.py3-none-any.whl
Collecting requests (from err)
  Using cached requests-2.7.0-py2.py3-none-any.whl
Collecting bottle (from err)
Collecting yapsy>=1.11 (from err)
Collecting backports.functools-lru-cache (from err)
Collecting jinja2 (from err)
  Using cached Jinja2-2.8-py2.py3-none-any.whl
Collecting markdown (from err)
  Using cached Markdown-2.6.2-py2.py3-none-any.whl
Collecting webtest (from err)
Collecting ansi (from err)
Collecting colorlog (from err)
  Using cached colorlog-2.6.0-py2.py3-none-any.whl
Collecting Pygments>=2.0.2 (from err)
  Using cached Pygments-2.0.2-py2-none-any.whl
Collecting config (from err)
Collecting dnspython (from err)
Collecting six>=1.5.2 (from pyOpenSSL->err)
  Using cached six-1.9.0-py2.py3-none-any.whl
Collecting cryptography>=0.7 (from pyOpenSSL->err)
Collecting MarkupSafe (from jinja2->err)
Collecting beautifulsoup4 (from webtest->err)
  Using cached beautifulsoup4-4.4.0-py2-none-any.whl
Collecting WebOb>=1.2 (from webtest->err)
Collecting waitress>=0.8.5 (from webtest->err)
  Using cached waitress-0.8.10-py2-none-any.whl
Collecting enum34 (from cryptography>=0.7->pyOpenSSL->err)
Collecting ipaddress (from cryptography>=0.7->pyOpenSSL->err)
  Using cached ipaddress-1.0.14-py27-none-any.whl
Collecting pyasn1>=0.1.8 (from cryptography>=0.7->pyOpenSSL->err)
Collecting idna>=2.0 (from cryptography>=0.7->pyOpenSSL->err)
  Using cached idna-2.0-py2.py3-none-any.whl
Collecting cffi>=1.1.0 (from cryptography>=0.7->pyOpenSSL->err)
Collecting pycparser (from cffi>=1.1.0->cryptography>=0.7->pyOpenSSL->err)
Installing collected packages: daemonize, Pygments, pygments-markdown-lexer, six, enum34, ipaddress, pyasn1, idna, pycparser, cffi, cryptography, pyOpenSSL, requests, bottle, yapsy, backports.functools-lru-cache, MarkupSafe, jin
ja2, markdown, beautifulsoup4, WebOb, waitress, webtest, ansi, colorlog, config, dnspython, err
Successfully installed MarkupSafe-0.23 Pygments-2.0.2 WebOb-1.4.1 ansi-0.1.3 backports.functools-lru-cache-1.0.3 beautifulsoup4-4.4.0 bottle-0.12.8 cffi-1.2.1 colorlog-2.6.0 config-0.3.9 cryptography-1.0.1 daemonize-2.3.1 dnspyt
hon-1.12.0 enum34-1.0.4 err-3.0.4 idna-2.0 ipaddress-1.0.14 jinja2-2.8 markdown-2.6.2 pyOpenSSL-0.15.1 pyasn1-0.1.8 pycparser-2.14 pygments-markdown-lexer-0.1.0.dev39 requests-2.7.0 six-1.9.0 waitress-0.8.10 webtest-2.0.18 yapsy
-1.11.223
/tmp/py2/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:90: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cau
se certain SSL connections to fail. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
(py2)$ cp lib/python2.7/s
site-packages/     site.pyc           sre_compile.pyc    sre_constants.pyc  sre_parse.pyc      stat.py            
site.py            sre_compile.py     sre_constants.py   sre_parse.py       sre.py             stat.pyc           
(py2)$ cp lib/python2.7/site
site-packages/ site.py        site.pyc       
(py2)$ cp lib/python2.7/site
site-packages/ site.py        site.pyc       
(py2)$ cp lib/python2.7/site-packages/errbot/config-template.py config.py
(py2)$ err.py -c config.py                                                                                                       
You are trying to run err under python2 without converting the source code to py2 first.
Either use python3 or install err using ./setup.py develop.
(py2)$ pip --version
pip 7.1.2 from /tmp/py2/lib/python2.7/site-packages (python 2.7)
(py2)$ 
```

To be honest, I do not get it either.

**Comment by gbin** [comment]
(py2)➜  ~  virtualenv --version                                                                                                                                                            
1.11.4

I am late 12 versions ? O o
What is your OS?

**Comment by mrshu** [comment]
@gbin I am on Arch Linux so it might be the beeding edge but everyone will get version 13 at some point.

**Comment by gbin** [comment]
I just reproduced it on arch.
```
py2)minisal ➜  ~  mkvirtualenv -p `which python2`
(py2)minisal ➜  ~  python --version
Python 2.7.10
(py2)minisal ➜  ~  pip --version
pip 7.1.2 from /home/gbin/.virtualenvs/py2/lib/python2.7/site-packages (python 2.7)
(py2)minisal ➜  ~ pip install err 
```

**Comment by gbin** [comment]
I released a 3.0.5, it should work now. Thanks for your patience and precise reporting !

**Comment by eykd** [comment]
Confirmed, it works for me now, too. Thanks!
